src/components/ScheduleWeek.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <v-container
+  >
+    <!--exercises receives from parent-->
+    Понедельник
+    <v-row
+      class="d-flex align-stretch"
+    >
+      <v-col>
+        <schedule-day
+        :dayExercises="evenDaysSchedule[0]"
+        />
+      </v-col>
+      <v-col>
+        <schedule-day
+        :dayExercises="oddDaysSchedule[0]"
+        />
+      </v-col>
+    </v-row>
+    Вторник
+    <v-row
+      class="d-flex align-stretch"
+    >
+      <v-col>
+        <schedule-day
+        :dayExercises="evenDaysSchedule[1]"
+        />
+      </v-col>
+      <v-col>
+        <schedule-day
+        :dayExercises="oddDaysSchedule[1]"
+        />
+      </v-col>
+    </v-row>
+    Среда
+    <v-row
+      class="d-flex align-stretch"
+    >
+      <v-col>
+        <schedule-day
+        :dayExercises="evenDaysSchedule[2]"
+        />
+      </v-col>
+      <v-col>
+        <schedule-day
+        :dayExercises="oddDaysSchedule[2]"
+        />
+      </v-col>
+    </v-row>
+    Четверг
+    <v-row
+      class="d-flex align-stretch"
+    >
+      <v-col>
+        <schedule-day
+        :dayExercises="evenDaysSchedule[3]"
+        />
+      </v-col>
+      <v-col>
+        <schedule-day
+        :dayExercises="oddDaysSchedule[3]"
+        />
+      </v-col>
+    </v-row>
+    Пятница
+    <v-row
+      class="d-flex align-stretch"
+    >
+      <v-col>
+        <schedule-day
+        :dayExercises="evenDaysSchedule[4]"
+        />
+      </v-col>
+      <v-col>
+        <schedule-day
+        :dayExercises="oddDaysSchedule[4]"
+        />
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+<script lang="ts">
+import Vue from "vue";
+import { Component, Prop } from "vue-property-decorator";
+import { IExercise } from "@/interfaces"
+import ScheduleDay from '@/components/ScheduleDay.vue'
+
+@Component({
+  name: 'ScheduleWeek',
+  components: {
+    ScheduleDay,
+  }
+})
+export default class ScheduleWeek extends Vue {
+  @Prop( {required: true, type: Array } ) readonly exercises!: IExercise[];
+  
+
+  created(){
+    this.parity = "2";
+    let orderedExercises = this.orderedItems(this.exercises)
+    for (let i = 0; i != orderedExercises.length; ++i){
+      this.setDaysSchedule(orderedExercises[i], "!2");
+    }
+    this.parity = "1";
+    orderedExercises = this.orderedItems(this.exercises)
+    for (let i = 0; i != orderedExercises.length; ++i){
+      this.setDaysSchedule(orderedExercises[i], "!1");
+    }
+  };
+
+  evenDaysSchedule: Array<Array<IExercise>> = [[],[],[],[],[]]; // chetn
+  oddDaysSchedule: Array<Array<IExercise>> = [[],[],[],[],[]];
+
+  counter = 0;
+
+  setDaysSchedule(exercise: IExercise, parity: string) {
+    switch (parity) {
```

Comment:
Можно таким образом попробовать :-)
```suggestion
    let daysSchedule;
switch (parity){
	case "!2":
    	daysSchedule = this.evenDaysSchedule;
    case "!1":
    	daysSchedule = this.oddDaysSchedule;
}
let dayIdx = parseInt(exercise.day) - 1;
if (daysSchedule && dayIdx >=0 && dayIdx < 5)
  daysSchedule.push(exercise)
```

Reply:
Немного не понял как это должно работать: подразумевается, что в конце функции мы по условию присвоим либо `this.evenDaysSchedule`, либо `this.oddDaysSchedule` значение `daysSchedule`?

Reply:
Именно, референс сохранится в `daysSchedule`.

Reply:
Это точно будет выгоднее?
```
    let daysShedule;
    switch(parity){
      case "!2":
        daysShedule = this.evenDaysSchedule;
        break;
      case "!1":
        daysShedule = this.oddDaysSchedule;
        break;
    }
    let dayIdx = parseInt(exercise.day) - 1;
    if (daysShedule && dayIdx >= 0 && dayIdx <= 5){
      daysShedule[dayIdx].push(exercise)
    }
    switch(parity){
      case "!2":
        this.evenDaysSchedule.push(daysShedule[0]);
        break;
```
Сначала мы объявляем двумерный массив, а потом используем единственный его элемент для присвоения. Ф-я, в которой вызывается этот код, используется для каждого эл-а расписания 
```
let orderedExercises = this.orderedItems(this.exercises)
for (let i = 0; i != orderedExercises.length; ++i){
   this.setDaysSchedule(orderedExercises[i], "!2");
}
```

Reply:
Я думаю, что если и есть разница в перформансе - она пренебрежимо мала, особенно учитывая размер `orderedExercises.length`, взамен этого код получается короче.